spec/dockerfile_spec.rb

```diff
@@ -0,0 +1,170 @@
+#! /usr/bin/env ruby
+
+require 'rspec/core'
+require 'json'
+
+describe 'The docker-compose file works' do
+  def which(cmd)
+    exts = ENV['PATHEXT'] ? ENV['PATHEXT'].split(';') : ['']
+    ENV['PATH'].split(File::PATH_SEPARATOR).each do |path|
+      exts.each { |ext|
+        exe = File.join(path, "#{cmd}#{ext}")
+        return exe if File.executable?(exe) && !File.directory?(exe)
+      }
+    end
+    return nil
+  end
+
+  def puppetserver_health_check(container)
+    %x(#{@docker} inspect "#{container}" --format '{{.State.Health.Status}}').chomp
+  end
+
+  def get_puppetdb_state
+    status = %x(#{@compose} exec -T puppet curl -s 'http://puppetdb:8080/status/v1/services/puppetdb-status').chomp
+    return JSON.parse(status)['state'] unless status.empty?
+    return ''
+  end
+
+  def start_puppetserver
+    container = %x(#{@compose} ps -q puppet).chomp
+    while container.empty?
+      sleep(1)
+      container = %x(#{@compose} ps -q puppet).chomp
+    end
+    status = puppetserver_health_check(container)
+    while status == 'starting'
+      sleep(1)
+      status = puppetserver_health_check(container)
+    end
+
+    # work around SERVER-2354
+    %x(#{@compose} exec puppet puppet config set server puppet)
+
+    return status
+  end
+
+  def start_puppetdb
+    status = get_puppetdb_state
+    while status != 'running'
+      sleep(1)
+      status = get_puppetdb_state
+    end
+    return status
+  end
+
+  def run_agent(agent_name)
+    %x(#{@docker} run --rm -it --net pupperware_default --name #{agent_name} --hostname #{agent_name} puppet/puppet-agent-ubuntu)
+    return $?
+  end
+
+  def check_report(agent_name)
+    domain = %x(#{@compose} exec -T puppet facter domain).chomp
+    body = "{ \"query\": \"nodes { certname = \\\"#{agent_name}.#{domain}\\\" } \" }"
+    out = ''
+    while out.empty?
+      out = %x(#{@compose} exec -T puppet curl -s -X POST http://puppetdb:8080/pdb/query/v4 -H 'Content-Type:application/json' -d '#{body}')
```

Comment:
Same for curl...

`-s` => `--silent`
`-X` => `--request`
`-H` => `--header`
`-d` => `--data`